h/static/scripts/hooks/use-group-annotations.tsx

```diff
@@ -111,6 +121,35 @@ export function useGroupAnnotations({
     [],
   );
 
+  // Used to cancel currently in-flight requests when this is unmounted
+  const abortControllers = useRef(new Set<AbortController>());
```

Comment:
To check my understand of where we're at, under the current plan we'll prevent there from being multiple requests in flight at once, so we don't actually need this to be a `Set`?

If so, what should `loadAnnotationsForPage` do if due to a bug there is another request in flight? Ignore it?